velox/serializers/PrestoSerializer.cpp

```diff
@@ -822,6 +822,84 @@ class VectorStream {
     return children_[index].get();
   }
 
+  // Similiar as flush(OutputStream* out)
+  size_t serializedSize() {
```

Comment:
I'm a bit concerned about how to ensure this method stays in sync with 'flush'. I wonder if it would be possible to implement this method by calling flush with an instance of OutputBuffer that simply counts number of written bytes.

Reply:
Yeah, this is a very good idea. I could add a new class to extend OutputStream, but rewrite the function `write` to only record the size. How about `OutputBufferSize : public OutputStream`, and then the function `flushSize` is not needed.
Do you agree this?

Reply:
Place the `OutputBufferSize` in ByteStream.h or PrestoSerializer.cpp?

Reply:
I'm thinking about something like this:

```
class CountingOutputStream : public OutputStream {
 public:
  explicit CountingOutputStream() : OutputStream{nullptr} {}

  void write(const char* s, std::streamsize count) override {
    pos_ += count;
    if (numBytes_ < pos_) {
      numBytes_ = pos_;
    }
  }

  std::streampos tellp() const override {
    return pos_;
  }

  void seekp(std::streampos pos) override {
    pos_ = pos;
  }

  std::streamsize size() const {
    return numBytes_;
  }

 private:
  std::streamsize numBytes_{0};
  std::streampos pos_{0};
};

  size_t serializedSize() {
    CountingOutputStream out;
    flush(&out);
    return out.size();
  }
```

Reply:
CountingOutputStream could be a class in anonymous namespace in PrestoSerializer.cpp

Reply:
Changed, thanks for your advice.